gcc/rust/feature/rust-feature.cc

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2020-2022 Free Software Foundation, Inc.
+
+// This file is part of GCC.
+
+// GCC is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the Free
+// Software Foundation; either version 3, or (at your option) any later
+// version.
+
+// GCC is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+// for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with GCC; see the file COPYING3.  If not see
+// <http://www.gnu.org/licenses/>.
+
+#include "rust-session-manager.h"
+#include "ordered-hash-map.h"
+#include "rust-feature.h"
+
+namespace Rust {
+
+Feature feature_table[] = {
+
+  DECLARE_ACTIVE_FEATURE ("associated_type_bounds", "1.34.0", "0.0.0", 52662, 0,
+			  ""),
+  DECLARE_ACTIVE_FEATURE ("intrinsics", "1.0.0", "0.0.0", 0, 0, ""),
+  DECLARE_ACTIVE_FEATURE ("rustc_attrs", "1.0.0", "0.0.0", 0, 0, ""),
```

Comment:
What is edition `0`? If I understand correctly, this is the equivalent of `Option<Edition>` used in rustc. In that case, we can use `Optional<Edition>` :) the class is defined in `gcc/rust/util/rust-optional.h` and should help you here